module/eCampLib/src/Service/ServiceUtils.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace eCamp\Lib\Service;
+
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\ORMException;
+use Doctrine\ORM\QueryBuilder;
+use eCamp\Lib\Acl\Acl;
+use eCamp\Lib\Acl\NoAccessException;
+use eCamp\Lib\Entity\BaseEntity;
+use eCamp\Lib\EntityFilter\EntityFilterInterface;
+use eCamp\Lib\ServiceManager\EntityFilterManager;
+use Zend\Hydrator\HydratorInterface;
+use Zend\Hydrator\HydratorPluginManager;
+use Zend\Permissions\Acl\Resource\ResourceInterface;
+use Zend\Permissions\Acl\Role\RoleInterface;
+
+class ServiceUtils
```

Comment:
Lösung mit ServieUtils finde ich für den Moment ok.

Die Frage ist, ob wir den Umweg via ServiceUtils wirklich gehen wollen, dem ServiceUtil die gesamten entityFilterManager und hydratorPluginManager zu injecten, welcher dann ja pro Entity nur wieder genau einen konkreten filter und hydrator herausliest. 

Mögliche Alternativlösung wär, dem Service bereits die konkreten filter und hydrator zu injecten (siehe Beispiel in https://github.com/usu/ecamp3/commit/de885b4589bd8a05b13caad6e5f2aa65500c2b69).

Vorteil:
- klareres Interface. Es ist eindeutig dass der EventCategoryService vom EventCategoryHydrator abhängt. keine versteckten Abhängigkeiten.
- geringere Code Komplexität (keine zusätzlichen manager notwendig)
- einfacheres Testing/Mocking: ich muss kein hydratorManager darum herum bauen

Nachteil:
- mehr Abhängigkeiten im constructor

Von mir aus können wir diese Diskussion aber von diesem Pull request lösen und separat anschauen. Mir ist lieber der Di-Code ist früher im Master, dann können weitere features darauf aufbauen.

Reply:
Grösster Nachteil:
Wenn die Basis-Klasse der Services eine zusätzliche Abhängigkeit bekommt, müssen alle Service-Klassen angepasst werden - daher hab ich dies in eine Zwischen-Stufe ausgelagert.

Alle anderen Inputs find ich gut.